For bug #78, add RelativeList to DataManager view to separate text...

dizzley opened the pull request:
…and checkbox. Add ellipsize.

This will prevent long instance names from crashing into the checkbox in the DataManagerList. (See screen grab in issue #78) I used a RelativeLayout to place the checkbox to the right of the two TextViews. Additionally, I added a maxLines of 2 and an ellipsis at the end for items longer than two lines. I think this gives a good user experience. Only a layout change was needed, no Java changes.

Thoughts?

![Effect of the fix](https://cloud.githubusercontent.com/assets/791168/20867674/f85f9f60-ba41-11e6-8995-cb963830252b.PNG)


_Edited for typo_

peterh-uk committed:
For bug #78, add RelativeList to DataManager view to separate text and checkbox. Add ellipsize.

dizzley commented:
Note - for some reason my Github username for the PR was @clamdrive - an old name of mine.

lognaturel commented:
Sweeeet, thanks, @dizzley for tackling this long-standing problem. Since the issue was filed, instance names have become more useful so it's even more important that they are shown properly. Instead of truncating at two lines, could you please let the instance name take up as many lines as it needs? Thanks! It's now possible to define a [template for generating the names](http://xlsform.org/#instance-name) and often the unique part is at the end.

We don't yet have automatic checks for style but are currently working towards it. While you're in this file, could you please make sure spaces are used everywhere and that the indentation is consistent throughout? You can see from [the diff](https://github.com/opendatakit/collect/pull/282/files) where the issues are and that indentation was inconsistent before you got in there. If you're in Android Studio, you can probably right click the file and run `Reformat Code`.

dizzley commented:
@lognaturel Yeah, that all makes sense. I'll carry on. (At the day job right now). 

By the way there will be a useful natural limit for the length of an instance name but happy to leave it un-truncated. You can ellipsize in the middle like this: "extremely long, long...g, long name" but will leave that feature out for now. (Deletes one line in XML file).

peterh-uk committed:
Add sublime edit project files to .gitignore

peterh-uk committed:
Fixes #78. Format instancename and checkbox correctly for long instance names.
Also removeunwanted truncation and ellipsize of instance name in two_item_multiple_choice layout.

dizzley commented:
Updated PR with suggestions from @lognaturel . Ran Reformat Code on generated layout XML.

lognaturel commented:
🎉 Awesome! @yanokwa please merge.

For next time, @dizzley, we're trying to keep PRs super focused so something like changing the .gitignore would work best as its own separate one. That means easier rollbacks for the community and more PRs for you! 😄

yanokwa commented:
Congrats on your first pull request, @dizzley and looking forward to your next one.

Fixes #78.

dizzley commented:
@lognaturel Thanks. My first PR here. re: .gitignore, Yeah I know... I just wanted my local fork to be neat. Next time I'll hold it back.

lognaturel commented:
@dizzley Oh no, I mean if you see small things like that which are not exactly related to the feature/fix you're working on, go ahead and do a separate PR. It's simple enough it can get merged right away and then it keeps branches focused on just one thing.
